pangeo_forge_recipes/transforms.py

```diff
@@ -638,3 +642,100 @@ def expand(
         # consolidate metadata and/or coordinate dims here
 
         return singleton_target_store
+
+
+@dataclass
+class CreatePyramid(beam.PTransform):
+    level: int
+    epsg_code: Optional[str] = None
+    rename_spatial_dims: Optional[dict] = None
+    pyramid_kwargs: Optional[dict] = field(default_factory=dict)
+
+    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
+        return pcoll | beam.Map(
+            create_pyramid,
+            level=self.level,
+            epsg_code=self.epsg_code,
+            rename_spatial_dims=self.rename_spatial_dims,
+            pyramid_kwargs=self.pyramid_kwargs,
+        )
+
+
+@dataclass
+class StoreToPyramid(beam.PTransform, ZarrWriterMixin):
+
+    # Open design question. Should StoreToPyramid repeat all
+    # of the StoreToZarr kwargs and pass them through to the nested StoreToZarr
+    # or should it packaged them as store_to_zarr_kwargs and unpack?
+
+    """Store a PCollection of Xarray datasets as a Zarr Pyramid.
+    :param n_levels: Number of pyramid levels to generate
+    :param combine_dims: The dimensions to combine
+    :param store_name: Zarr store will be created with this name under ``target_root``.
+    :param other_chunks: Chunks for non-spatial dims.
+    Spatial dims are controlled by pixels_per_tile. Default is None
+      If a dimension is a not named, the chunks will be inferred from the data.
+    :param target_root: Root path the Zarr store will be created inside;
+        `store_name` will be appended to this prefix to create a full path.
+    :param epsg_code: EPSG code to set dataset CRS if CRS missing. If provided will overwrite.
+        Default is None.
+    :param rename_spatial_dims: Dict containing the new spatial dim names for Rioxarray.
+        Default is None.
+    :param pyramid_kwargs: Dict containing any kwargs that should be passed to ndpyramid.
+        Default is None.
+
```

Comment:
Eventually we should add tests for the two currently supported projections. Since this is only relevant for one of the methods, `pyramid_reproject` I think it should go in `pyramid_kwargs` rather than a separate kwarg.